Kinvey/Kinvey/HttpRequestFactory.swift

```diff
@@ -426,21 +426,22 @@ struct HttpAppDataRequestFactory: AppDataRequestFactory {
     func buildAppDataSave<T: Persistable, Result>(
         _ persistable: T,
         options: Options?,
+        isNew: Bool?,
         resultType: Result.Type
     ) -> HttpRequest<Result> {
         let collectionName = try! T.collectionName()
         let client = options?.client ?? self.client
         var bodyObject = try! client.jsonParser.toJSON(persistable)
         let objId = bodyObject[Entity.EntityCodingKeys.entityId] as? String
-        let isNewObj = isNew(entityId: objId)
+        let isNewObj = isNew ?? Kinvey.isNew(entityId: objId)
         let request = HttpRequest<Result>(
             httpMethod: isNewObj ? .post : .put,
             endpoint: isNewObj ? AppDataEndpoint.appData(client: client, collectionName: collectionName) : AppDataEndpoint.appDataById(client: client, collectionName: collectionName, id: objId!),
             credential: client.activeUser,
             options: options
         )
         
-        if isNewObj {
+        if isNewObj || (Kinvey.isNew(entityId: objId)) {
```

Comment:
If we are defining `isNewObj` above to set it to `Kinvey.isNew(...)` if it is initially null, what is the reason for checking it here again?

Reply:
you are right, removing it now!

Reply:
thanks for notice this